lib/project_types/script/layers/infrastructure/languages/assemblyscript_task_runner.rb

```diff
@@ -49,12 +49,31 @@ def metadata
 
           def library_version(library_name)
             output = JSON.parse(CommandRunner.new(ctx: ctx).call("npm list --json"))
-            raise Errors::APILibraryNotFoundError.new(library_name), output unless output["dependencies"][library_name]
-            output["dependencies"][library_name]["version"]
+            library_version_from_npm_list(output, library_name)
+          rescue Errors::SystemCallFailureError => error
+            library_version_from_npm_list_error(error, library_name)
           end
 
           private
 
+          def library_version_from_npm_list_error(error, library_name)
```

Comment:
My only nit is on this naming. In this code base we usually suffix our errors with `error`, so this method makes me think that we're returning an error. However, we're returning the library version if it exists.

I'm bad with naming but what if we did `library_version_from_npm_list_error_output` or something?